pkg/domain/blueprintSpec.go

```diff
@@ -245,20 +246,29 @@ func (spec *BlueprintSpec) DetermineStateDiff(installedDogus map[common.SimpleDo
 		return nil // do not re-determine the state diff from status StatusPhaseStateDiffDetermined and above
 	}
 
-	doguDiffs := determineDoguDiffs(spec.EffectiveBlueprint.Dogus, installedDogus)
-	compDiffs, err := determineComponentDiffs(spec.EffectiveBlueprint.Components, installedComponents)
+	doguDiffs := determineDoguDiffs(spec.EffectiveBlueprint.Dogus, ecosystemState.InstalledDogus)
+	compDiffs, err := determineComponentDiffs(spec.EffectiveBlueprint.Components, ecosystemState.InstalledComponents)
 	if err != nil {
+		//FIXME: a proper state and event should be set, so that this error don't lead to an endless retry.
```

Comment:
Fix it

Reply:
Can we do that in a separate branch? This could lead to a bigger refactoring (especially for the tests) and this code is not related to this PR.